src/clusterfuzz/_internal/bot/fuzzers/centipede/engine.py

```diff
@@ -88,6 +90,32 @@ class Engine(engine.Engine):
   def name(self):
     return 'centipede'
 
+  def _get_arguments(self, fuzzer_path):
+    """Gets the fuzzer arguments.
+    This function will fetch for an existing `.options` file and add some
+    default arguments.
+    Args:
+      fuzzer_path: Path to the main target in a string.
+
+    Returns:
+      A list of arguments.
+    """
+    arguments = fuzzer_options.FuzzerArguments({})
+    options = fuzzer_options.get_fuzz_target_options(fuzzer_path)
+
+    if options:
+      arguments = options.get_engine_arguments('centipede')
+
+    # We ignore this parameter in the options file because it doesn't really
+    # make sense not to crash on errors.
+    arguments['exit_on_crash'] = 1
+
+    for key, val in _DEFAULT_ARGUMENTS.items():
+      if key not in arguments:
+        arguments[key] = val
```

Comment:
Consider `arguments.setdefault(key, val)`.

Reply:
`arguments` is not an actual dictionary, it's a custom object that vaguely behaves like one. I first wanted to clean the implementation of it, but that's not the purpose of this CL, so I prefer leaving things like it.